### PR TITLE
seo(docs): metadata refactor - use frontmatter

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

**Issue:** after https://github.com/aws-amplify/amplify-ui/pull/1563, we use meta info from the `data/meta` file, but it's hard to maintain.

**Solution**: In this PR, we get `metaTitle` and `metaDescription` from each `.mdx` file's frontmatter, so that it's hard to forget to update. 

Also, add the `supportedFramework` for sitemap generation (https://github.com/aws-amplify/amplify-ui/pull/1900) and future dynamic framework switcher button.

Note: 
- `/` and `/[framework]` pages are not `.tsx` files, which doesn't have frontmatter, so we have to keep using `data/meta` file
- in the future, if we want to make the `metaTitle` and `metaDescription` dynamic, we can use `[platform]` in its verbiage.

#### Issue #, if available

https://app.asana.com/0/1201840995372760/1202276284378545/f

#### Description of how you validated changes

manually validated on local

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by joebuono** [comment]
It looks like there's quite a bit of redundancy in each `.mdx` file's frontmatter. In other words, `title` is the same as `metaTitle`, and `description` is the same as `metaDescription`. 

Perhaps there is a way to pluck the content from `title` and `description` without duplicating the content in `metaTitle` and `metaDescription`?

**Comment by 0618** [comment]
> It looks like there's quite a bit of redundancy in each `.mdx` file's frontmatter. In other words, the content for `title` is the same as the content for `metaTitle`, and the same goes for `description` and `metaDescription`.
> 
> Perhaps there is a way to pluck the content from `title` and `description` without duplicating the content in `metaTitle` and `metaDescription`?

Reduce redundancy is generally good, but in this case we are doing it on purpose so that we can have more control over the visible text and invisible info

File: docs/src/pages/[platform]/guides/auth-protected/auth-protected.vue.mdx [hunk 1]
```diff
@@ -1 +0,0 @@
-## Coming Soon
```
Comment (reesscot) [inline docs/src/pages/[platform]/guides/auth-protected/auth-protected.vue.mdx:1]: 💯

File: docs/src/pages/[platform]/theming/responsive/index.page.mdx [hunk 2]
```diff
@@ -1,5 +1,8 @@
 ---
 title: Responsive Design
+metaTitle: Responsive Design
+metaDescription: Responsive styling is supported out of the box using our default breakpoints. Our responsive support uses a mobile first approach, so @media(min-width) rules are used for all breakpoints.
```
Comment (reesscot) [inline docs/src/pages/[platform]/theming/responsive/index.page.mdx:4]:
```suggestion
metaDescription: Amplify UI supports responsive design out of the box. Build responsive layouts in your applications using responsive style props with the built-in set of breakpoints, write custom media queries in CSS, or use the Flex and Grid components.
```

**Comment by reesscot** [comment]
> > It looks like there's quite a bit of redundancy in each `.mdx` file's frontmatter. In other words, the content for `title` is the same as the content for `metaTitle`, and the same goes for `description` and `metaDescription`.
> > Perhaps there is a way to pluck the content from `title` and `description` without duplicating the content in `metaTitle` and `metaDescription`?
> 
> Reduce redundancy is generally good, but in this case we are doing it on purpose so that we can have more control over the visible text and invisible info

+1, our next step will be to customize the meta descriptions for better search engine optimization. We don't necessarily want to have the same text for the description displayed to the user as the text we put on our search listings.

**Review by reesscot — APPROVED** [review]
LGTM

**Review by joebuono — APPROVED** [review]

**Pull request merged**